When user tries to edit query by using the edit icon, modify the quer…

marcosholgado opened the pull request:
…y rather than do a new search.

<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.
-->

Task/Issue URL: https://app.asana.com/0/1125189844152671/1185449784809161
Tech Design URL: 
CC: 

**Description**:
This PR changes the behaviour of the edit search icon in the system search. Right now when a user clicks on the icon it triggers a new search and the query input doesn't change . The purpose of the icon is to edit the query so the user can further edit it (as the app's omnibar behaves).

**Steps to test this PR**:
1. Add the app widget and open it.
1. Do a search.
1. Click on the icon with an arrow in the right side of a search result.
1. It should change the query in the omnibar to the result selected and move the caret to the end.

**Item selectable widget**
1. Add the app widget and open it.
1. Do a search.
1. Click on the icon with an arrow in the right side of a search result.
1. Notice how the arrow view has a selectable background that highlights the user is clicking it.

**Item selectable normal search**
1. Launch the app without using the widget.
1. Do a search.
1. Click on the icon with an arrow in the right side of a search result.
1. Notice how the arrow view has a selectable background that highlights the user is clicking it.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

marcosholgado commented:
@malmstein, @subsymbolic I've put you both down for reviewing this small PR, there is no rush though. I think @subsymbolic is probably the best reviewer for this but I've added @malmstein as well in case you get a chance to have a look at it. Again, no rush, is just a small fix that was bothering me while on vacation :D.

subsymbolic commented:
@marcosholgado thanks for submitting this, I will review tomorrow.

subsymbolic commented:
@malmstein want me to reassign this PR review to you? Looks like you're already in there.

malmstein commented:
Yes @subsymbolic, happy to help!

marcosholgado commented:
> Love the improvement @marcosholgado. Could we add a `android:background="?selectableItemBackground"` to the view? It's not clear that it's a clickable icon. 
>

Nice improvement! I've added that change and updated the description to test that change as well 👍
